src/cmd/sign.js

```diff
@@ -125,6 +156,170 @@ export default function sign(
   );
 }
 
+/**
+ * Fetches an updateManifest.json file from a remote web server
+ */
+function fetchUpdateManifest(manifestData: Object): Object {
+  return new Promise(async function (resolve, /*reject*/) {
+    let oldUpdateManifest;
+    let statusCode;
+
+    let parsed = url.parse(manifestData.applications.gecko.update_url);
+    let updateManifestFileName = '';
+    // these will never be null, since parsed and parsed.pathname
+    // are already validated in an earlier function
+    // the `if` is only to please flow
+    if (parsed != null && parsed.pathname != null) {
+      updateManifestFileName = path.basename(parsed.pathname);
+    }
+
+    try {
+      [oldUpdateManifest, statusCode] = await httpFetchFile(
+        manifestData.applications.gecko.update_url);
+      if (statusCode < 200 || statusCode > 290) {
+        throw new WebExtError(
+          `Failed to retrieve ${updateManifestFileName}\n` +
+          `http statusCode error, server responded with: ${statusCode}`);
+      }
+    } catch (e) {
+      throw new WebExtError(
+        `Was unable to download ${updateManifestFileName}\nerror is: ${e}`);
+    }
+
+    try {
+      resolve(JSON.parse(oldUpdateManifest));
+    } catch (e) {
+      throw new WebExtError(
+        `Unable to parse ${updateManifestFileName} file located at ` +
+        `${manifestData.applications.gecko.update_url}`);
+    }
+  });
+}
+
+/**
+ * Validates all data needed to generate a new updatemanifest
+ * if some data is missing/corrupt, the manifest can't be generated
+ */
+function prevalidateUpdateManifestParams(manifestData, updateLink) {
+  // ..first validate if the manifest has the right properties
+  // manifestData.applications.gecko.updateUrl
+  // note: `var == null` is equal to
+  // `(variable === undefined || variable === null)`
+  // see: https://stackoverflow.com/questions/2647867
+  if (manifestData == null ||
```

Comment:
yeah, sorry, I have no idea what I was talking about either. I guess ignore it for now. I will do another review when you've addressed the other change requests anyway.